### PR TITLE
Motivate and explain regular expression rules.

### PR DESCRIPTION
**Pull request by timoreimann**
@nikore what do you think, is this good enough?

@containous/traefik PTAL.

### CODE REVIEW

**Comment by nikore** [comment]
Looks great; thanks!

**Comment by trecloux** [comment]
LGTM 👍

File: docs/basics.md [hunk 1]
```diff
@@ -85,13 +85,17 @@ Frontends can be defined using the following rules:
 - `PathStrip`: Same as `Path` but strip the given prefix from the request URL's Path.
 - `PathPrefix`: PathPrefix adds a matcher for the URL path prefixes. This matches if the given template is a prefix of the full URL path.
 - `PathPrefixStrip`: Same as `PathPrefix` but strip the given prefix from the request URL's Path.
-- `AddPrefix` : Add prefix from the request URL's Path.
+- `AddPrefix`: Add prefix from the request URL's Path.
```
Comment (dtomcej) [inline docs/basics.md]: Sorry grammar question, shouldn't it be:

 `AddPrefix`: Add prefix to the request URL's Path. ?

Comment (timoreimann) [inline docs/basics.md]: Absolutely!

**Comment by timoreimann** [comment]
@dtomcej okay to merge?

**Pull request merged**